include/libmpu/mpu6050.hpp

```diff
@@ -1,5 +1,223 @@
 #pragma once
 
+#include <libhal-util/bit.hpp>
+#include <libhal-util/i2c.hpp>
+#include <libhal-util/map.hpp>
+#include <libhal/accelerometer.hpp>
+
 namespace hal::mpu {
+class mpu6050 : public hal::accelerometer
+{
+public:
+  /// The device address when A0 is connected to GND.
+  static constexpr hal::byte address_ground = 0b110'1000;
+  /// The device address when A0 is connected to V+.
+  static constexpr hal::byte address_voltage_high = 0b110'1001;
+  /// The device address when A0 is connected to SDA.
+  static constexpr hal::byte address_sda = 0b100'1010;
+  /// The device address when A0 is connected to SCL.
+  static constexpr hal::byte address_scl = 0b100'1011;
+
+  /// The address of the read-only register containing the temperature data.
+  static constexpr hal::byte xyz_register = 0x3B;
+  /// The address of the register used to configure gravity scale the device.
+  static constexpr hal::byte configuration_register = 0x1C;
+  /// The address of the register used to initilize the device.
+  static constexpr hal::byte initalizing_register = 0x6B;
+
+  /// The command to enable one-shot shutdown mode.
+  static constexpr hal::byte who_am_i_register = 0x75;
+  
+  /**
+   * @brief Constructs and returns MPU object
+   * 
+   * @param i2c I2C bus the MPU is connected to 
+   * @param device_address 0x68 if address pin is pulled low, 0x69 if pulled high
+   * @return result<mpu6050> 
+   */
+  static result<mpu6050> create(hal::i2c& i2c,
```

Comment:
```suggestion
  /**
   * @brief Constructs and returns MPU object
   * 
   * @param i2c - I2C bus the MPU is connected to 
   * @param device_address - 0x68 if address pin is pulled low, 0x69 if pulled high
   * @return result<mpu6050> 
   */
  static result<mpu6050> create(hal::i2c& i2c,
```

Add a dash separator between parameter name and the definition. Just something we do for better readability when looking at the code.